### ISSUE TITLE
Docker compose build fails due to missing composer.json in ./app (es_php_fpm)

### ISSUE THREAD

**Issue by llasse** [engelsystem/engelsystem#822]
On CentOS7  when running Docker ( Client: Docker Engine - Community Version 19.03.12 / Server: Docker Engine - Community  Version:   20.10.2 ) I encounter the following error when trying to build the docker image:

```
$ sudo docker-compose build
es_database uses an image, skipping
Building es_php_fpm
Sending build context to Docker daemon  18.94kB
Step 1/30 : FROM composer AS composer
 ---> 84d313cd1a80
Step 2/30 : COPY ./ /app/
 ---> Using cache
 ---> da3c0ed25c4e
Step 3/30 : RUN composer --no-ansi install --no-dev --ignore-platform-reqs
 ---> Running in f5e8eb9a1909
Composer could not find a composer.json file in /app
To initialize a project, please create a composer.json file as described in the https://getcomposer.org/ "Getting Started" section
The command '/bin/sh -c composer --no-ansi install --no-dev --ignore-platform-reqs' returned a non-zero code: 1
ERROR: Service 'es_php_fpm' failed to build : Build failed
```

Any suggestions on how to troubleshoot / solve this? ( I tried messing with the es_php_fpm Dockerfile but couldn't solve it by changing the order of the steps (as it seems to be copied some steps later)

**Comment by MyIgel** [comment]
Hi, that's interesting and "works for me" i assume you are building the "normal" engelsystem and not the "dev" engelsystem? Are you in the docker directory?
Your output shows `18.94kB`, the checked out repo and zip downloads should be around 3MB -> are you working on some release file or a repo zip download? Or is there somewhere a restrictive .dockerignore outside of the repo?

tldr: Your problem here seems to be that docker does not read all files when building the image.

**Comment by llasse** [comment]
Steps I took:
Downloaded https://github.com/engelsystem/engelsystem/archive/refs/tags/v3.2.0.tar.gz via wget (I assumed that this would be the right file to install via docker?)
unpacked the tar, went into engelsystem-3.2.0/docker and ran sudo docker compose build

Maybe something went wrong here, already?

I do have other docker containers running (mostly linuxserver.io ones..) but the corresponding compose-file is in another folder in my home directory and should thus not affect it..

**Comment by MyIgel** [comment]
So long that is a valid approach and other containers should not affect it.

The expected output is:
```
[~]$ wget https://github.com/engelsystem/engelsystem/archive/refs/tags/v3.2.0.tar.gz
--2021-07-25 22:34:19--  https://github.com/engelsystem/engelsystem/archive/refs/tags/v3.2.0.tar.gz
[...]
2021-07-25 22:34:20 (6,05 MB/s) - ‘v3.2.0.tar.gz’ saved [724378]

[~]$ tar -xzf v3.2.0.tar.gz 
[~]$ cd engelsystem-3.2.0/docker/
[docker]$ docker-compose build
es_database uses an image, skipping
Building es_php_fpm
Sending build context to Docker daemon  3.057MB
Step 1/30 : FROM composer AS composer
 ---> c69d012e5132
Step 2/30 : COPY ./ /app/
 ---> Using cache
 ---> 5ecb8af1252e
Step 3/30 : RUN composer --no-ansi install --no-dev --ignore-platform-reqs
 ---> Running in 9d6a4d9139ac
No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
Loading composer repositories with package information
Updating dependencies
Lock file operations: 93 installs, 0 updates, 0 removals
  - Locking dms/phpunit-arraysubset-asserts (v0.2.1)
  - Locking doctrine/cache (2.1.1)
  - Locking doctrine/dbal (2.13.2)
  - Locking doctrine/deprecations (v0.5.3)
  - Locking doctrine/event-manager (1.1.1)
[...]
```
-> does it work when you run the above commands from for example `/tmp`? Are you shure that all files got extracted and that they are in the `engelsystem-3.2.0` directory? Are there eventually some file system permissions problems? As i mentioned earlier:Your build output suggests that most of the files where not copied to the build env and thus are missing.

**Comment by msquare** [comment]
@llasse were you able to resolve the problem?

**Comment by MyIgel** [comment]
@llasse I'll close that issue for now. If you have further questions or if it does not work for you, feel free to re-open it :)
